chatbot/main.py

```diff
@@ -0,0 +1,63 @@
+## import libraries
+import os
+from vectordb import ChatbotAgent
+from load_data import load_data
+
+
+
+# main function
+def main():
+	## OpenAI API Key
+	with open(r'chatbot\OpenAI-API-key\OpenAI_API_key.txt', 'r') as f:
+		api_key = f.read().strip()
+	os.environ['OPENAI_API_KEY'] = api_key
+	print("OpenAI API Key Set\n")
+
+	## Configure Chroma
+	persist_directory = r'chatbot\vector-db-persist-directory\chroma\chatbot'
+
+	# URLs of the files to be merged from Project Open-academy
+	_sources_urls = load_data()
+
+	## Initialize the ChatbotAgent
+	chatbot_agent = ChatbotAgent(_openai_api_key=os.environ["OPENAI_API_KEY"], _sources_urls=_sources_urls, _persist_directory=persist_directory)
+	
+	print("\n\n****Chatbot Agent Initialized****")
+	# start the conversation
+	while 1:

```

Comment:
How could you launch an endless program in Github action?

Reply:
In fact in Github action, it does not run mian.py. The action in workflows is used to update the MD links and store them (links) in  /chatbot/vector-db-persist-directory every time they submit a PR or submit. The action does not train the data but modifies/transforms the data.

As the output is printed to the terminal windows, the user makes the continuous conversation. So it uses a "while loop".